tests/frontend/test_BoardGUI.py

```diff
@@ -1,73 +1,177 @@
+
+
+@patch('pygame.mouse.get_pos', return_value=(100, 100))
+@patch('pygame.mouse.get_pressed', return_value=(1, 0, 0))
+def test_get_cell_clicked(mock_get_pressed, mock_get_pos, board_gui_fixture):
+    board_gui, _ = board_gui_fixture
+    cell = board_gui.get_cell_clicked()
+    expected_cell = pg.Vector2(100, 100) // board_gui.cell_size
+    assert cell == expected_cell
+    mock_get_pos.assert_called_once()
+    mock_get_pressed.assert_called_once()
+
+
+@patch('pygame.mouse.get_pos', return_value=(100, 100))
+@patch('pygame.mouse.get_pressed', return_value=(0, 1, 0))  # Simulate a right-click
+def test_get_cell_clicked_with_right_click(mock_get_pressed, mock_get_pos, board_gui_fixture):
+    board_gui, _ = board_gui_fixture
+    cell = board_gui.get_cell_clicked()
+    
+    # Expect no cell to be returned on right-click
+    assert cell is None  
+    mock_get_pos.assert_called_once()
+    mock_get_pressed.assert_called_once()
+
+@patch('pygame.mouse.get_pos', return_value=(100, 100))
+@patch('pygame.mouse.get_pressed', return_value=(0, 0, 0))  # Simulate no mouse click
+def test_get_cell_clicked_with_no_click(mock_get_pressed, mock_get_pos, board_gui_fixture):
+    board_gui, _ = board_gui_fixture
+    cell = board_gui.get_cell_clicked()
+    
+
+    assert cell is None  
+    mock_get_pos.assert_called_once()
+    mock_get_pressed.assert_called_once()
+    
+def test_draw_piece(board_gui_fixture):
+    board_gui, mock_game = board_gui_fixture
+    position = (100, 100)
+
+    # Test drawing a black piece
+    board_gui.draw_piece('black', position)
+    mock_game.screen.blit.assert_called_once_with(board_gui.black_piece_image, position)
+
+    # Assert that the black piece is at the correct position on the screen
+    assert piece_exists_at_position(mock_game.screen, board_gui.black_piece_image, position), "Black piece not found at the expected position"
+
+    # Reset mock and test drawing a white piece
+    mock_game.screen.blit.reset_mock()
+    board_gui.draw_piece('white', position)
+    mock_game.screen.blit.assert_called_once_with(board_gui.white_piece_image, position)
+
+    # Assert that the white piece is at the correct position on the screen
+    assert piece_exists_at_position(mock_game.screen, board_gui.white_piece_image, position), "White piece not found at the expected position"
+
+
+def piece_exists_at_position(screen, piece_image, position):
+    """Helper function to check if a piece exists at a specific position."""
+    # Here you might need to implement logic that checks if the image is rendered at the given position.
+    # Since this is a mock, we will check the call arguments for blit
+    calls = screen.blit.call_args_list
+    for call in calls:
+        if call[0][0] == piece_image and call[0][1] == position:
+            return True
+    return False
+
+
+
+# This test case failed because logic in backend is not correct to throw error
+def test_draw_piece_with_invalid_position(board_gui_fixture):
```

Comment:
I think if we cant get it updated by tuesday we should adjust this test to reflect the current code. So that its not reflected poorly on us in grading.